packages/protocol/contracts/layer1/core/impl/Inbox.sol

```diff
@@ -458,11 +465,56 @@ contract Inbox is IInbox, IForcedInclusionStore, EssentialContract {
 
     /// @dev Builds and persists transition records for batch proof submissions
     /// Validates transitions, calculates bond instructions, and stores records
-    /// @dev Virtual function that can be overridden for optimization (e.g., transition aggregation)
+    /// @dev Aggregates consecutive proposals to reduce storage and event costs.
```

Comment:
@ggonzalez94 maybe we overlooked a potential optimization for this function in the first place -- why don't we enable the off-chain prover to prove aggregated proofs across multiple consecutive proposals, by **adding `span` directly into the `Transition` record**. Then we don't need to work on aggregation on-chain.